boa/util/abi.py

```diff
@@ -1,23 +1,31 @@
 # wrapper module around whatever encoder we are using
+from typing import Any
 
 from eth.codecs.abi.decoder import Decoder
 from eth.codecs.abi.encoder import Encoder
+from eth.codecs.abi.nodes import ABITypeNode
 from eth.codecs.abi.parser import Parser
+from eth_abi import is_encodable
 
-_parsers: dict[str, Parser] = {}
+_parsers: dict[str, ABITypeNode] = {}
 
 
-def _get_parser(schema):
+def _get_parser(schema: str):
     try:
         return _parsers[schema]
     except KeyError:
         _parsers[schema] = (ret := Parser.parse(schema))
         return ret
 
 
-def abi_encode(schema, data):
+def abi_encode(schema: str, data: Any) -> bytes:
     return Encoder.encode(_get_parser(schema), data)
 
 
-def abi_decode(schema, data):
+def abi_decode(schema: str, data: bytes) -> Any:
     return Decoder.decode(_get_parser(schema), data)
+
+
+# todo: eth.codecs.abi does not have such a function, which one do we use?
```

Comment:
i think it's ok. we could also try `abi_encode` and catch any exceptions, but i think it ends up being about the same thing